Lib/fontbakery/profiles/outline.py

```diff
@@ -0,0 +1,227 @@
+                continue
+            for i in range(0, len(segments)):
+                prev = segments[i - 1]
+                this = segments[i]
+                if len(prev) == 2 and len(this) == 2:
+                    if (
+                        abs(prev.tangentAtTime(0).angle - this.tangentAtTime(0).angle)
+                        < COLINEAR_EPSILON
+                    ):
+                        warnings.append(f"{glyphname}: {prev} -> {this}")
+
+    if warnings:
+        formatted_list = "\t* " + pretty_print_list(list(set(warnings)), sep="\n\t* ")
+        yield WARN, Message(
+            "found-colinear-vectors",
+            f"The following glyphs have colinear vectors:\n" f"{formatted_list}",
+        )
+    else:
+        yield PASS, ("No colinear vectors found.")
+
+
+@check(
+    id="com.google.fonts/check/outline_jaggy_segments",
+    conditions=["outlines_dict", "is_not_variable_font"],
+)
+def com_google_fonts_check_outline_jaggy_segments(ttFont, outlines_dict):
+    """Do outlines contain any jaggy segments?"""
+    warnings = []
+    for glyphname, outlines in outlines_dict.items():
+        for p in outlines:
+            segments = p.asSegments()
+            outline_area = None
+            if not segments:
+                continue
+            for i in range(0, len(segments)):
+                prev = segments[i - 1]
+                this = segments[i]
+                jag_angle = math.acos(
+                    ((prev.start - prev.end) @ (this.end - this.start))
+                    / (
+                        (prev.start - prev.end).magnitude
+                        * (this.end - this.start).magnitude
+                    )
+                )
+                if abs(jag_angle) > JAG_ANGLE:
+                    continue
+                jag_area = (
+                    ((prev.start.x * prev.end.y) - (prev.start.y * prev.end.x))
+                    + ((this.start.x * this.end.y) - (this.start.y * this.end.x))
+                    + ((prev.start.x * this.end.y) - (prev.start.y * this.end.x))
+                )
+                if jag_area < JAG_AREA_EPSILON * p.area:
+                    warnings.append(f"{glyphname}: {prev}/{this}")
+
+    if warnings:
+        formatted_list = "\t* " + pretty_print_list(list(set(warnings)), sep="\n\t* ")
+        yield WARN, Message(
+            "found-jaggy-segments",
+            f"The following glyphs have jaggy segments:\n" f"{formatted_list}",
+        )
+    else:
+        yield PASS, ("No jaggy segments found.")
+
+
+@check(
+    id="com.google.fonts/check/outline_semi_vertical",
+    conditions=["outlines_dict", "is_not_variable_font"],
+)
+def com_google_fonts_check_outline_semi_vertical(ttFont, outlines_dict):
+    """Do outlines contain any semi-vertical or semi-horizontal lines?"""
+    warnings = []
+    for glyphname, outlines in outlines_dict.items():
+        for p in outlines:
+            segments = p.asSegments()
+            outline_area = None
```

Comment:
Travis noticed this variable is defined but not used. And the same thing for another var with the same name at line 147:
![Screenshot from 2020-11-09 15-29-50](https://user-images.githubusercontent.com/213676/98581737-b1ead100-22a0-11eb-97ea-a3cea70e7388.png)
https://travis-ci.org/github/googlefonts/fontbakery/jobs/742453976